src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,43 @@
+package duke;
+
+/**
+ * Represents the Main Duke program which has-a Storage, TaskList and Ui.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
```

Comment:
```suggestion
    private final Storage storage;
    private final TaskList tasks;
    private final Ui ui;
```
Suggesting the addition of `final` modifier.

Reply:
I did not implement it as final as I may need to modify the said variable later on.